### PR TITLE
Add 'manage' statement to language spec

### PR DESCRIPTION
**Pull request by dlongnecke-cray**
Add 'manage' statement to language spec (#19478)

Add a small section on the `manage` statement to the `Statements`
chapter of the language spec.

TESTING

- [x] `make spectests`
- [x] Viewed updated docs on web

Reviewed by @lydia-duncan, @arezaii. Thanks!

Signed-off-by: David Longnecker <dlongnecke-cray@users.noreply.github.com>

### CODE REVIEW

**Comment by dlongnecke-cray** [comment]
Thanks @arezaii! @lydia-duncan would you be willing to do a pass for wordsmithing? I figure something is lacking 😄.

File: doc/rst/language/spec/statements.rst [hunk 1]
```diff
@@ -993,3 +996,227 @@ given by
...
+Participating types must also define the ``leaveThis()`` method,
+which is called implicitly when the scope of the managed block
+is exited.
+
+The ``leaveThis()`` method takes an ``Error?`` by ``in`` intent. If
+the error is not ``nil``,  it may be handled within the method. It
+can also be propagated by annotating ``leaveThis()`` with the
+``throws`` tag and throwing the error.
+
+Manager expressions may also be nested within a manage statement.
```
Comment (lydia-duncan) [inline doc/rst/language/spec/statements.rst:1160]: I tend to think of nesting as:
```chapel
manage wrapper1 as val1 {
  manage wrapper2 as val2 {
    ...
  }
}
```
It seems like an implementation detail that the code you've written counts as nested, though I think the ordering of entering and leaving is still important to convey.  I think I'd use similar wording to [multiple variable declarations](https://chapel-lang.org/docs/1.26/language/spec/variables.html#multiple-variable-declarations)

**Review by lydia-duncan — APPROVED** [review]
I think the only major thing for me is the quibble about "nesting", the other two are take-it-or-leave-it suggestions

File: doc/rst/language/spec/statements.rst (hunk 1 above)
Comment (dlongnecke-cray) [reply doc/rst/language/spec/statements.rst:1160]: Semantically the "nesting" is equivalent to the code you've written here. I'll give that link a look. Maybe something like "multiple manager expressions may be present within a single manage statement".

**Review by lydia-duncan — APPROVED** [review]
Sweet, thanks!

**Pull request merged**